core/src/replay_stage.rs

```diff
@@ -3077,48 +3077,51 @@ impl ReplayStage {
                 }
 
                 if bank.collector_id() != my_pubkey {
-                    // If the block does not have at least DATA_SHREDS_PER_FEC_BLOCK shreds in the last FEC set,
-                    // mark it dead. No reason to perform this check on our leader block.
-                    if !blockstore
-                        .is_last_fec_set_full(bank.slot())
-                        .inspect_err(|e| {
+                    // If the block does not have at least DATA_SHREDS_PER_FEC_BLOCK correctly retransmitted
```

Comment:
Outside the scope of this review, but it would be nice to sub-function this (and other areas of this giant function). Something like:
```
if !last_fec_set_is_valid(bank, blockstore, etc.) {
    continue;
}

fn last_fec_set_is_valid(..) {
    all this code to check and possibly mark slot dead
}
```

Reply:
i agree, will look to refactor this on master in a future pr